src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -33,162 +43,465 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+
+        internal static DiagnosticDescriptor MethodTypeParameterMoreMeaningfulNameRule = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            s_localizableMessageMethodTypeParameterMoreMeaningfulName,
+            DiagnosticCategory.Naming,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: false,
+            description: s_localizableDescription,
+            helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/ca1704",
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            FileParseRule,
+            AssemblyRule,
+            NamespaceRule,
+            TypeRule,
+            VariableRule,
+            MemberRule,
+            MemberParameterRule,
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule);
+
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private void OnCompilationStart(CompilationStartAnalysisContext compilationStartContext)
+        {
+            var dictionaries = ReadDictionaries();
+            var projectDictionary = CodeAnalysisDictionary.CreateFromDictionaries(dictionaries.Concat(_mainDictionary));
+
+            compilationStartContext.RegisterOperationAction(AnalyzeVariable, OperationKind.VariableDeclarator);
+            compilationStartContext.RegisterCompilationEndAction(AnalyzeAssembly);
+            compilationStartContext.RegisterSymbolAction(
+                AnalyzeSymbol,
+                SymbolKind.Namespace,
+                SymbolKind.NamedType,
+                SymbolKind.Method,
+                SymbolKind.Property,
+                SymbolKind.Event,
+                SymbolKind.Field,
+                SymbolKind.Parameter);
+
+            IEnumerable<CodeAnalysisDictionary> ReadDictionaries()
+            {
+                var fileProvider = AdditionalFileProvider.FromOptions(compilationStartContext.Options);
+                return fileProvider.GetMatchingFiles(@"(?:dictionary|custom).*?\.(?:xml|dic)$")
+                    .Select(CreateDictionaryFromAdditionalText)
```

Comment:
There is a performance hit for method group, I don't know if it matters a lot for analyzer.

```suggestion
                    .Select(x => CreateDictionaryFromAdditionalText(x))
```

Reply:
This is news to me; do you have some documentation? The answers on [this question](https://stackoverflow.com/questions/6662532/difference-between-lambda-expression-and-method-group) seem to suggest the opposite.

Reply:
Yeah that's a weird thing. Here is the a link to my [SO question](https://stackoverflow.com/q/53208516/4625433) and I know there was a good blog post (I think from JetBrains) but I can't seem to find it back...

Reply:
That is interesting, but based on your findings, we shouldn't cross the 1M iteration boundary where this might be a concern.